channelsubscriptions/channelsubscriptions.go

```diff
@@ -0,0 +1,51 @@
+package channelsubscriptions
+
+import (
+	"sync"
+
+	datatransfer "github.com/filecoin-project/go-data-transfer/v2"
+	"github.com/filecoin-project/go-data-transfer/v2/channels"
+)
+
+type ChannelSubscriptions struct {
+	subscriptions   map[datatransfer.ChannelID][]datatransfer.Subscriber
+	subscriptionsLk sync.RWMutex
+	unsub           datatransfer.Unsubscribe
+}
+
+type SubscriptionAPI interface {
+	SubscribeToEvents(subscriber datatransfer.Subscriber) datatransfer.Unsubscribe
+}
+
+func NewChannelSubscriptions(subscriptionAPI SubscriptionAPI) *ChannelSubscriptions {
+	cs := &ChannelSubscriptions{
+		subscriptions: make(map[datatransfer.ChannelID][]datatransfer.Subscriber),
+	}
+	cs.unsub = subscriptionAPI.SubscribeToEvents(cs.subscriber)
+	return cs
+}
+
+func (cs *ChannelSubscriptions) Stop() {
+	cs.unsub()
+}
+
+func (cs *ChannelSubscriptions) Subscribe(chid datatransfer.ChannelID, cb datatransfer.Subscriber) {
+	cs.subscriptionsLk.Lock()
+	defer cs.subscriptionsLk.Unlock()
+	cs.subscriptions[chid] = append(cs.subscriptions[chid], cb)
+}
+
+func (cs *ChannelSubscriptions) subscriber(evt datatransfer.Event, state datatransfer.ChannelState) {
```

Comment:
this feels a bit odd, could just be the name and the fact that it's for test only; was it meant to do more?